pallets/automation-time/src/lib.rs

```diff
@@ -245,6 +300,56 @@ pub mod pallet {
 			Ok(().into())
 		}
 
+		/// Schedule a task to fire an event with a custom message.
+		///
+		/// Before the task can be scheduled the task must past validation checks.
+		/// * The transaction is signed
+		/// * The provided_id's length > 0
+		/// * The time is valid
+		/// * 
+		///
+		/// # Parameters
+		/// * `provided_id`: An id provided by the user. This id must be unique for the user.
+		/// * `time`: The unix standard time in seconds for when the task should run.
+		/// * `message`: The message you want the event to have.
+		///
+		/// # Errors
+		/// * `InvalidTime`: Time must end in a whole minute.
+		/// * `PastTime`: Time must be in the future.
+		/// * `DuplicateTask`: There can be no duplicate tasks.
+		/// * `TimeSlotFull`: Time slot is full. No more tasks can be scheduled for this time.
+		/// * `InvalidAmount`: Amount has to be larger than 0.1 OAK.
+		/// * `InvalidRecipient`: Recipient must be a valid address.
+		/// * `TransferToSelf`: Sender cannot transfer money to self.
+		/// * `InsufficientFunds`: Amount in sender account is insufficient.
+		#[pallet::weight(<T as Config>::WeightInfo::schedule_transfer_task_existing_slot())]
+		pub fn schedule_transfer_task(
+			origin: OriginFor<T>,
+			provided_id: Vec<u8>,
+			time: UnixTime,
+			recipient_id: T:: AccountId,
+			amount: T::Balance,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			// check for greater than existential deposit
+			// ensure!(amount < T::ExistentialDeposit::get(), Err(<Error<T>>::InvalidAmount)?)?;
+			if amount < T::ExistentialDeposit::get() {
+				Err(<Error<T>>::InvalidAmount)?
+			}
+			// check not sent to self
+			if who != recipient_id {
+				Err(<Error<T>>::TransferToSelf)?
+			}
+			let senderBalance = <T as Config>::Currency::free_balance(&who);
+			if senderBalance < amount {
```

Comment:
types don't match. 

```senderBalance: <<T as pallet::Config>::Currency as frame_support::traits::Currency<<T as frame_system::Config>::AccountId>>::Balance;

```amount: <T as pallet::Config>::Balance;

my initial thought was to just adjust our T::Balance to be the same as Currency::Balance: https://github.com/OAK-Foundation/OAK-blockchain/pull/74/files#diff-5d19ac6ca496cd503aa07940ae2dc4a0fdffdfbb6c49726507a70c314135d7a0R80

but that's still not working because it's seemingly not recognizing the type that's being passed in as anything more than T::Balance, since it's not sure what's being injected during launch. so i'm not sure how to coerce the type to like what i'm passing in